HotelBookingTests/HotelExampleTwoStepPaymentTest.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using HotelBase;
+using HotelBooking;
+using NUnit.Framework;
+using Rhino.Mocks;
+
+namespace HotelBookingTests
```

Comment:
Should have:

In your 'architecture' you add a new class with full hotel booking operation flow e.g. book room, check price. However in your tests you have just limit to payment.  In my pinion you have two/three options:
* test whole class
* change your architecture, which may not be easy
* maybe also share some common test cases

Reply:
applies (tested whole flow)